client/cmd/import.go

```diff
@@ -0,0 +1,255 @@
+package cmd
+
+import (
+	"alda.io/client/code-generator"
+	"alda.io/client/help"
+	"alda.io/client/interop"
+	log "alda.io/client/logging"
+	"alda.io/client/model"
+	"alda.io/client/system"
+	"alda.io/client/transmitter"
+	"alda.io/client/util"
+	"fmt"
+	"github.com/logrusorgru/aurora"
+	"github.com/spf13/cobra"
+	"os"
+	"strings"
+)
+
+var outputAldaFilename string
+var importFormat string
+
+func init() {
+	importCmd.Flags().StringVarP(
+		&file, "file", "f", "", "Read data from a file to convert to Alda",
+	)
+
+	importCmd.Flags().StringVarP(
+		&code, "code", "c", "", "Read data from a string to convert to Alda",
+	)
+
+	importCmd.Flags().StringVarP(
+		&outputAldaFilename, "output", "o", "", "The output Alda code filename",
+	)
+
+	importCmd.Flags().StringVarP(
+		&importFormat, "import-format", "i", "", "The format of the imported data",
+	)
+}
+
+var importCmd = &cobra.Command{
+	Use:   "import",
+	Hidden: true,
+	Short: "Evaluate external format and import as Alda source code",
+	Long:  `Evaluate external format and import as Alda source code
+
+---
+
+Source code can be provided in one of three ways:
+
+The path to a file (-f, --file):
+  alda import -i musicxml -f path/to/my-score.musicxml -o my-score.alda
+
+A string of code (-c, --code):
+  alda import -i musicxml -c "...some musicxml data..." -o my-score.alda
+
+Text piped into the process on stdin:
+  echo "...some musicxml data..." | alda import -i musicxml -o my-score.alda
+
+---
+
+When -o / --output FILENAME is provided, the results are written into that file.
+
+  alda import -i musicxml -c "...some musicxml data..." -o my-score.alda
+
+Otherwise, the results are written to stdout, which is convenient for
+redirecting into other files or processes.
+
+  alda import -i musicxml -f path/to/my-score.musicxml > my-score.alda
+  alda import -i musicxml -f path/to/my-score.musicxml | some-process > my-score.alda
+
+---
+
+Currently, the only import format is MusicXML.  
+
+---`,
+	RunE:  func(_ *cobra.Command, args[]string) error {
+		if importFormat != "musicxml" {
+			return help.UserFacingErrorf(
+				`%s is not a supported input format.
+
+Currently, the only supported output format is %s.`,
+				aurora.BrightYellow(importFormat),
+				aurora.BrightYellow("musicxml"),
+			)
+		}
+
+		// TODO (experimental): remove warning log
+		log.Warn().Err(help.UserFacingErrorf(
+			`The %s command is currently experimental. Imported scores may be incorrect and lack information.`,
+			aurora.BrightYellow("import"),
+		)).Msg("Experimental command")
```

Comment:
It would be simpler to log a string here instead of an error, something like:

```go
log.Warn().
  Str(
   "detail",
   fmt.Sprintf(
     `The %s command is currently experimental. Imported scores may be incorrect and lack information.`,
     aurora.BrightYellow("import"))).
  Msg("Experimental command")
```

Or maybe even:

```go
log.Warn().
  Msg(
   fmt.Sprintf(
     `The %s command is currently experimental. Imported scores may be incorrect and lack information.`,
     aurora.BrightYellow("import")))
```

Reply:
Great this is a lot better. Just defaulted to creating an error. Resolved.